### PR TITLE
samples: led_ws2812: Add esp32xx overlays

### PR DESCRIPTION
**Pull request by marekmatej**
Add overlays for `esp32s2`, `esp32s3` and `esp32c3` based boards, which have RGB led assembled. Those boards are capable of driving the RGB led using SPI, but the default MOSI signal is assigned differently to how RGB led `DI` is wired on the boards.
This PR depends on this https://github.com/zephyrproject-rtos/zephyr/pull/60183 in order to work.

### CODE REVIEW

File: samples/drivers/led_ws2812/boards/esp32c3_devkitm.overlay [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2022 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/dt-bindings/led/led.h>
+
+&spi2 {
```
Comment (sylvioalves) [inline samples/drivers/led_ws2812/boards/esp32c3_devkitm.overlay:9]: @marekmatej, can you add `line-idle-low;` in all `spi`entries after #60183?

Comment (beriberikix) [reply samples/drivers/led_ws2812/boards/esp32c3_devkitm.overlay:9]: I needed to add this to make it work for the esp32c3 I tested.

**Comment by simonguinot** [comment]
Hi @marekmatej. Is it OK to be reviewed and merged now ?

**Comment by marekmatej** [comment]
> Hi @marekmatej. Is it OK to be reviewed and merged now ?

Hi @simonguinot let me rebase and test it.

**Review by simonguinot — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Review by beriberikix — APPROVED** [review]
Verified against:

* S3 devkitc-1 v1.0
* C3 devkitm-1 v1.0
* C3 devkitc-02 v1.1

All worked great!

**Pull request merged**